From 6548e580509397a622b7c504a79de93414771459 Mon Sep 17 00:00:00 2001
From: Christian Marangi <ansuelsmth@gmail.com>
Date: Wed, 25 Jun 2025 00:04:36 +0200
Subject: [PATCH 6/6] net: ethernet: airoha: define sport value for GDM3

On Airoha AN7583, the Serdes Ethernet goes through the GDM3 port.
To correctly receive packet for QDMA, add the sport value to identify
packet from GDM3 port.

Signed-off-by: Christian Marangi <ansuelsmth@gmail.com>
---
 drivers/net/ethernet/airoha/airoha_eth.c | 3 +++
 1 file changed, 3 insertions(+)

--- a/drivers/net/ethernet/airoha/airoha_eth.c
+++ b/drivers/net/ethernet/airoha/airoha_eth.c
@@ -604,6 +604,9 @@ static int airoha_qdma_get_gdm_port(stru
 	case 0x18:
 		port = 3; /* GDM4 */
 		break;
+	case 0x16:
+		port = 2; /* GDM3 */
+		break;
 	case 0x10 ... 0x14:
 		port = 0; /* GDM1 */
 		break;
